code/modules/mod/modules/modules_supply.dm

```diff
@@ -212,6 +183,46 @@
 		ores -= ore
 	drain_power(use_power_cost)
 
+/obj/item/mod/module/plasma_engine
+	name = "MOD plasma engine module"
+	desc = "A minaturized plasma engine, capable of directly intaking raw and refined plasma to recharge the MODsuit's core in the field."
+	icon_state = "module"
+	module_type = MODULE_TOGGLE
+	complexity = 3
+	active_power_cost = 0
+	incompatible_modules = list(/obj/item/mod/module/plasma_engine)
+	cooldown_time = 0.5 SECONDS
+	allowed_inactive = TRUE
+	var/list/charger_list = list(/obj/item/stack/ore/plasma = 1500, /obj/item/stack/sheet/mineral/plasma = 3000)
+
+/obj/item/mod/module/plasma_engine/on_activation()
+	. = ..()
+	RegisterSignal(mod, COMSIG_PARENT_ATTACKBY, PROC_REF(on_attackby))
+	balloon_alert(mod.wearer, "engine Online, insert plasma into core unit.")
```

Comment:
Should be chat messages rather than balloon alerts, maybe paired with a sound. Same for deactivation.
 
 This honestly goes for almost all of the balloon alerts, they are overused in tg's code and end up being quite ugly.